daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -599,6 +599,7 @@ func (devices *DeviceSet) cleanupDeletedDevices() error {
 
 	// If there are no deleted devices, there is nothing to do.
 	if devices.nrDeletedDevices == 0 {
+		devices.Unlock()
```

Comment:
@LK4D4 

Are you saying do "defer devices.Unlock()" right after taking lock? I can't do that and reason being that with-in same function I need to drop the lock before calling devices.DeleteDevice(). And DeleteDevice() assumes that it will be called without devices.Lock() held.